scio-hdfs/src/main/scala/com/spotify/scio/hdfs/package.scala

```diff
@@ -158,16 +155,35 @@ package object hdfs {
       }
     }
 
-    private[scio] def hadoopDistCacheCopy(src: Path, target: URI, conf: Configuration): Unit = {
-      logger.debug(s"Will copy ${src.toUri}, to $target")
+    private def getTargetFromOptions() = {
+      if (self.options.getRunner() != classOf[DirectPipelineRunner]) {
+        val dfOptions = self.optionsAs[DataflowPipelineOptions]
+        require(self.optionsAs[DataflowPipelineOptions].getStagingLocation != null,
+          "Staging directory not set - use `--stagingLocation`!")
+        (dfOptions.getStagingLocation, gcsOutputStream _)
+      } else {
+        // should targetDir be specified in options?
```

Comment:
It seems like this code to lives in `com.google.cloud.dataflow.sdk.runners.DataflowPipelineRunner.fromOptions`, it handles making sure that staging or temp is set, and if needed sets the missing values. Will this code run after or before `fromOptions` ^ ?